donkeycar/parts/pytorch/torch_data.py

```diff
@@ -0,0 +1,149 @@
+# PyTorch
+import torch
+from torch.utils.data import IterableDataset, DataLoader
+from donkeycar.utils import train_test_split
+from donkeycar.parts.tub_v2 import Tub
+from torchvision import transforms
+from typing import List, Any
+from donkeycar.pipeline.types import TubRecord, TubDataset
+from donkeycar.pipeline.sequence import TubSequence
+import pytorch_lightning as pl
+
+def get_default_transform(for_video=False, for_inference=False):
+    """
+    Creates a default transform to work with torchvision models
+
+    Video transform:
+    All pre-trained models expect input images normalized in the same way, 
+    i.e. mini-batches of 3-channel RGB videos of shape (3 x T x H x W), 
+    where H and W are expected to be 112, and T is a number of video frames 
+    in a clip. The images have to be loaded in to a range of [0, 1] and 
+    then normalized using mean = [0.43216, 0.394666, 0.37645] and 
+    std = [0.22803, 0.22145, 0.216989].
+    """
+
+    mean = [0.485, 0.456, 0.406]
+    std = [0.229, 0.224, 0.225]
+    input_size = (224, 224)
+
+    if for_video:
+        mean = [0.43216, 0.394666, 0.37645]
+        std = [0.22803, 0.22145, 0.216989]
+        input_size = (112, 112)
+
+    transform = transforms.Compose([
+        transforms.Resize(input_size),
+        transforms.ToTensor(),
+        transforms.Normalize(mean=mean, std=std)
+    ])
+
+    return transform
+
+
+class TorchTubDataset(IterableDataset):
```

Comment:
:clap: This is so awesome

Reply:
@DocGarbanzo anything you want me to do here?

Reply:
Its a good idea to leave this class as-is because we can account for things like `multiprocessing` as we get closer to the ML frameworks. Its good for our `Iterator`s to be more "general purpose" and leave all the specificity to the ML frameworks.